### PR TITLE
Port KerasNLP GPT-2 text generation example to multi-backend

### PR DESCRIPTION
**Pull request by abheesht17**
https://keras.io/examples/generative/gpt2_text_generation_with_kerasnlp/

### CODE REVIEW

**Comment by abheesht17** [comment]
JAX: https://colab.research.google.com/drive/1sN1wz2T5GX2w6GUOLSJOT7WCsMXfuZ4I?usp=sharing (works fine)

Torch: https://colab.research.google.com/drive/1TRas0yvs5dchFcHdGuQjDhwVWEvZn4QQ?usp=sharing (funny bug: https://github.com/keras-team/keras-nlp/issues/1144)

TF: https://colab.research.google.com/drive/1aJkc_RjfGoFlXvWG89MZA-eHoubJg0Yo?usp=sharing (`generate()` bug: https://github.com/keras-team/keras-nlp/issues/1145

File: examples/generative/gpt2_text_generation_with_kerasnlp.py [hunk 1]
```diff
@@ -25,16 +25,23 @@
 """
 
 """
-## Install KerasNLP and Import Dependencies
+## Install KerasNLP, Choose Backend and Import Dependencies
+
+We can choose one of "tensorflow", "jax", "torch" as the backend. Let's go
```
Comment (mattdangerw) [inline examples/generative/gpt2_text_generation_with_kerasnlp.py:30]: We've written the follow example to run on top of [Keras Core](https://keras.io/keras_core/) and work with multiple backends. You can chance these guide to run on `"tensorflow"` or `"torch"` simply by updating the `KERAS_BACKEND` line below.

File: examples/generative/gpt2_text_generation_with_kerasnlp.py [hunk 2]
```diff
@@ -25,16 +25,23 @@
...
+ahead with JAX in this example.
 """
 
 """shell
 pip install -q keras-nlp
 """
 
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
```
Comment (mattdangerw) [inline examples/generative/gpt2_text_generation_with_kerasnlp.py:40]: Leave a comment maybe `os.environ["KERAS_BACKEND"] = "jax"  # or "tensorflow" or "torch"`

Comment (mattdangerw) [reply examples/generative/gpt2_text_generation_with_kerasnlp.py:40]: Also, still not sure if we would want this to default to `"jax"` or `"tensorflow"`. We should figure out how we want to approach generally.

File: examples/generative/gpt2_text_generation_with_kerasnlp.py [hunk 3]
```diff
@@ -25,16 +25,24 @@
 """
 
 """
-## Install KerasNLP and Import Dependencies
+## Install KerasNLP, Choose Backend and Import Dependencies
+
+We've written this example to run on top of Keras Core and it works with
```
Comment (mattdangerw) [inline examples/generative/gpt2_text_generation_with_kerasnlp.py:30]: I would rephrase slightly...

The following example uses [Keras Core](https://keras.io/keras_core/) to work in any of `tensorflow`, `jax` or `torch`. Support for Keras Core is baked into KerasNLP, simply change the `KERAS_BACKEND` environment variable below to change the backend you would like to use. We select the `jax` backend below.

**Review by mattdangerw — APPROVED** [review]
Looks good! Just one comment.

**Comment by abheesht17** [comment]
@fchollet - should this example be added to https://github.com/keras-team/keras-core as well?

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
